graph-builder/src/config/options.rs

```diff
@@ -0,0 +1,128 @@
+//! Options shared by CLI and TOML.
+
+use super::{AppSettings, Merge};
+use commons::{parse_params_set, parse_path_prefix};
+use std::collections::HashSet;
+use std::net::IpAddr;
+use std::path::PathBuf;
+use std::time::Duration;
+
+// TODO(lucab): drop all aliases after staging+production deployments
+// have been aligned on new flags.
+
+/// Status service options.
+#[derive(Debug, Deserialize, Serialize, StructOpt)]
+pub struct StatusOptions {
+    /// Address on which the status service will listen
+    #[structopt(name = "status_address", long = "status.address")]
+    pub address: Option<IpAddr>,
+
+    /// Port to which the status service will bind
+    #[structopt(name = "status_port", long = "status.port")]
+    pub port: Option<u16>,
+}
+
+/// Options for the main Cincinnati service.
+#[derive(Debug, Deserialize, Serialize, StructOpt)]
+pub struct ServiceOptions {
+    /// Address on which the server will listen
+    #[structopt(name = "service_address", long = "service.address", alias = "address")]
+    pub address: Option<IpAddr>,
+
+    /// Port to which the server will bind
+    #[structopt(name = "service_port", long = "service.port", alias = "port")]
+    pub port: Option<u16>,
+
+    /// Namespace prefix for all service endpoints (e.g. '/<prefix>/v1/graph')
+    #[structopt(long = "service.path_prefix", parse(from_str = "parse_path_prefix"))]
+    pub path_prefix: Option<String>,
+
+    /// Comma-separated set of mandatory client parameters
+    #[structopt(
+        long = "service.mandatory_client_parameters",
+        parse(from_str = "parse_params_set")
+    )]
+    pub mandatory_client_parameters: Option<HashSet<String>>,
+}
+
+/// Options for the Docker-registry-v2 fetcher.
+#[derive(Debug, Deserialize, Serialize, StructOpt)]
+pub struct DockerRegistryOptions {
+    /// Duration of the pause (in seconds) between registry scans
+    #[structopt(
+        long = "upstream.registry.period",
+        parse(try_from_str = "duration_from_secs")
+    )]
+    #[serde(default = "Option::default", deserialize_with = "de_duration_secs")]
+    pub period: Option<Duration>,
+
+    /// URL for the container image registry
+    #[structopt(long = "upstream.registry.url", alias = "registry")]
+    pub url: Option<String>,
+
+    /// Name of the container image repository
+    #[structopt(long = "upstream.registry.repository", alias = "repository")]
+    pub repository: Option<String>,
+
+    /// Credentials file for authentication against the image registry
+    #[structopt(
+        long = "upstream.registry.credentials_path",
+        alias = "credentials-file"
+    )]
+    pub credentials_path: Option<PathBuf>,
```

Comment:
Self-note: this only works on toolchain >= 1.32.

Reply:
Split to https://github.com/openshift/cincinnati/pull/77.

Reply:
So a rebase should fix the build failure :+1: